### PR TITLE
refactor(app): adjust modal UI and create Confirm CL deletion modal

### LINKED ISSUE #10631: 6.0 Feedback:  [Design QA] Modal Dialogs #10167
### Overview

This is first round design QA on all the modals in our app. 

### Current Behavior

[Link to view the annotated frame in Figma. ](https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/Milestone-1%3A-Read-and-Execute?node-id=13869%3A276334)

**Why is this not available modal**
- [x] (1) Needs 1.5 rem margin beneath the button. Also remove the box shadow. 
Screen Shot 2022-06-06 at 12 19 40 PM

**How to set up a new robot**
- [x] (1) Remove the box-shadow from the button
- [x] (2) Needs 1.5 rem margin beneath the button. Also remove the box shadow

Screen Shot 2022-06-06 at 10 58 56 AM

**Delete this labware definition**
- [x] (1) Hasn't been implemented yet (including the background overlay update)
Screen Shot 2022-06-06 at 12 22 15 PM

**Factory settings modal**
- [x] (1) When this modal appears the breadcrumbs still appear above the background overlay
- [x] (2) Add a spacing of 1.5 rem between the 'cancel' and the button
- [x] (3) Have we updated the text here? Let me know and I'll update the designs 
- [x] (4) Hover state for this error version of the button isn't linked up! On hover it turns blue. Also remove the box shadow
- [x] (5) I have a sneaky feeling these toast icons haven't been updated 

Screen Shot 2022-06-06 at 4 28 23 PM

**Software Update Modal**
Don't know if this one has been updated?
Screen Shot 2022-06-06 at 4 29 23 PM
 

**How to restore to a previous software version modal**

- [x] (1) Needs 1.5 rem margin beneath the button
- [x] (2) Adjust the margins between the text to be .5rem (8px)
- [x] (3) Remove box shadow from button

Screen Shot 2022-06-06 at 4 33 15 PM

### PR DESCRIPTION
**Pull request by jerader**
closes #10631

# Overview

This PR tweaks a handful of modals mentioned in the ticket and creates the Confirm custom labware deletion modal

# Changelog

- added a new alert icon
- updated screenshot
- edit `Modal`
- create a Custom Labware Overflow menu test

# Review requests

- review figma, does it match designs?

# Risk assessment

low

### CODE REVIEW

File: app/src/organisms/Devices/RobotSettings/AdvancedTab/AdvancedTabSlideouts/FactoryResetModal.tsx [hunk 1]
```diff
@@ -77,27 +77,26 @@ export function FactoryResetModal({
...
               >
                 {t('shared:cancel')}
               </Link>
-              <PrimaryButton
-                backgroundColor={COLORS.error}
+              <AlertPrimaryButton
                 onClick={triggerReset}
                 disabled={PENDING_STATUS}
               >
                 {t('factory_reset_modal_confirm_button')}
```
Comment (koji) [inline app/src/organisms/Devices/RobotSettings/AdvancedTab/AdvancedTabSlideouts/FactoryResetModal.tsx:98]: Sorry, there was a miscommunication on this button.
The recent button label is `Yes, clear data and restart robot`

File: app/src/organisms/Labware/CustomLabwareOverflowMenu.tsx [hunk 2]
```diff
@@ -107,41 +121,36 @@ export function CustomLabwareOverflowMenu(
...
-          </AlertModal>
+            <Flex flexDirection={DIRECTION_COLUMN}>
+              <StyledText as="p">{t('def_moved_to_trash')}</StyledText>
+              <StyledText as="p" paddingTop={SPACING.spacing3}>
+                {t('cannot-run-python-missing-labware')}
+              </StyledText>
+              <Flex
+                justifyContent={JUSTIFY_FLEX_END}
+                alignItems={ALIGN_CENTER}
+                marginTop="24px"
```
Comment (koji) [inline app/src/organisms/Labware/CustomLabwareOverflowMenu.tsx]: Let's use `SPACING.spacing5`

File: app/src/organisms/Labware/CustomLabwareOverflowMenu.tsx [hunk 3]
```diff
@@ -107,41 +121,36 @@ export function CustomLabwareOverflowMenu(
...
+              >
+                <Btn
+                  onClick={handleCancelModal}
+                  textTransform={TEXT_TRANSFORM_CAPITALIZE}
+                  marginRight={SPACING.spacing5}
+                  css={TYPOGRAPHY.linkPSemiBold}
+                  fontWeight={TYPOGRAPHY.fontWeightSemiBold}
```
Comment (koji) [inline app/src/organisms/Labware/CustomLabwareOverflowMenu.tsx]: `TYPOGRAPHY.linkPSemiBold` includes `fontWeightSemiBold` so we don't need this.

File: app/src/atoms/Modal/index.tsx [hunk 4]
```diff
@@ -53,12 +53,12 @@ export const Modal = (props: ModalProps): JSX.Element => {
...
           justifyContent={JUSTIFY_SPACE_BETWEEN}
           paddingX={SPACING.spacing5}
-          paddingY={SPACING.spacing4}
+          paddingY={SPACING.spacing5}
         >
           <Flex>
             {['error', 'warning'].includes(type) ? (
               <Icon
-                name="alert-circle"
+                name="ot-alert"
```
Comment (koji) [inline app/src/atoms/Modal/index.tsx:61]: Do you know the difference between `alert-circle` and `ot-alert`?
I don't find the diff 😂

**Review by koji — APPROVED** [review]
Looks good to me!
I left a couple of comments but they are small things.
I think if we could get one more reviewer, it would be good.

File: app/src/organisms/Labware/CustomLabwareOverflowMenu.tsx (hunk 2 above)
Comment (jerader) [reply app/src/organisms/Labware/CustomLabwareOverflowMenu.tsx]: oh yup thanks! I forgot that I didn't convert that to our SPACINGs

File: app/src/atoms/Modal/index.tsx (hunk 4 above)
Comment (jerader) [reply app/src/atoms/Modal/index.tsx:61]: haha, one of Mel's AC + me staring at the icon convinced me that we didn't add the new alert icon. Maybe there is not a difference, I will ask around to see what others think 😅

**Review by vabruzzo — APPROVED** [review]
looks good to me 👍

**Comment by mmencarelli** [comment]
r2 feedback on modal's 

**Modal-wide edits**

- [x] (1) Change the padding of `.nkZbU` to: 1rem 1.5rem; (16px top and bottom, 24px r/left). 
- [x] (2) Change the padding of `.hJTtDI` to: 1rem 1.5rem 1.5rem 1.5rem; 
Screen Shot 2022-06-22 at 9 42 56 AM

**Why is this robot unavailable?**
- [x] (1) Missing ")." at the end of content!
Screen Shot 2022-06-22 at 12 30 21 PM


**Delete labware def**
- [x] (1) remove the box-shadow from button on hover
Screen Shot 2022-06-22 at 9 43 01 AM

**Reset to factory settings**
- [x] (1) Woops! Add in 1.5rem spacing between content and button
- [x] (2) remove box-shadow from button on hover

Screen Shot 2022-06-22 at 9 44 56 AM

**Review by mmencarelli — DISMISSED** [review]
Changes requested above

**Comment by jerader** [comment]
The round 2 work will be completed in this ticket: https://github.com/Opentrons/opentrons/issues/10866

**Pull request merged**